lib/puppet_x/templates/invoke_ps_command.erb

```diff
@@ -0,0 +1,225 @@
+    {
+      this.exitCode = 0;
+      this.shouldExit = false;
+    }
+
+    public override Guid InstanceId
+    {
+        get { return _hostId; }
+    }
+
+    public override string Name
+    {
+        get { return "PuppetPSHost"; }
+    }
+
+    public override Version Version
+    {
+        get { return new Version(1, 0); }
+    }
+
+    public override PSHostUserInterface UI
+    {
+        get { return null; }
+    }
+
+    public override CultureInfo CurrentCulture
+    {
+        get { return Thread.CurrentThread.CurrentCulture; }
+    }
+
+    public override CultureInfo CurrentUICulture
+    {
+        get { return Thread.CurrentThread.CurrentUICulture; }
+    }
+
+    public override void EnterNestedPrompt()
+    {
+        throw new NotImplementedException();
+    }
+
+    public override void ExitNestedPrompt()
+    {
+        throw new NotImplementedException();
+    }
+
+    public override void NotifyBeginApplication()
+    {
+        return;
+    }
+
+    public override void NotifyEndApplication()
+    {
+        return;
+    }
+
+    public override void SetShouldExit(int exitCode)
+    {
+      this.shouldExit = true;
+      this.exitCode = exitCode;
+    }
+}
+"@
+
+function Format-Result
+{
+  param($output, $exitcode, $errormessage)
+
+  if($PSVersionTable.PSVersion -gt [Version]'3.0'){
+    @{
+      stdout = "$($output)"
+      exitcode = $exitcode
+      errormessage = "$($errormessage)"
+    } | ConvertTo-Json -Compress
+  }else{
+@"
+{
+  "stdout": "$($output)",
+  "exitcode": $exitcode,
+  "errormessage": "$($errormessage)"
+}
+"@
+  }
+
+}
+
+function New-XmlResult
+{
+  param($exitcode, $output, $errormessage)
+
+  # we make our own xml because ConvertTo-Xml makes hard to parse xml
+  [System.Xml.XmlDocument]$xDoc = New-Object System.Xml.XmlDocument
+  [System.Xml.XmlElement]$xRoot = $xDoc.CreateElement("ReturnResult")
+  $xDoc.AppendChild($xRoot) | Out-Null
+
+  [System.Xml.XmlElement]$xElement = $xDoc.CreateElement("Property")
+  $xElement.SetAttribute('Name','exitcode')
+  $xElement.innerText = "$($exitCode)"
+  $xRoot.AppendChild($xElement) | Out-Null
+
+  [System.Xml.XmlElement]$xElement = $xDoc.CreateElement("Property")
+  $xElement.SetAttribute('Name','errormessage')
+  $xElement.innerText = $errormessage
+  $xRoot.AppendChild($xElement) | Out-Null
+
+  [System.Xml.XmlElement]$xElement = $xDoc.CreateElement("Property")
+  $xElement.SetAttribute('Name','stdout')
+  $cdata = $xdoc.CreateCDataSection("$($output)")
+  $xElement.AppendChild($cdata) | Out-Null
+  $xRoot.AppendChild($xElement) | Out-Null
+
+  $memStream = New-Object System.IO.MemoryStream
+  $writerSettings = New-Object System.Xml.XmlWriterSettings
+  $writerSettings.Indent = $true
+  $writerSettings.Encoding = New-Object System.Text.UTF8Encoding -Arg $false
+  $writerSettings.ConformanceLevel = [System.Xml.ConformanceLevel]::Document
+
+  $writer = [System.Xml.XmlWriter]::Create($memStream, $writerSettings)
+  $xDoc.Save($writer)
```

Comment:
This is still throwing an exception for the output from our buffer overlfow test. Despite having UTF8 specified everywhere I can think of